sql/src/main/java/org/apache/druid/sql/calcite/external/ExternalTableScanRule.java

```diff
@@ -43,17 +43,26 @@ public ExternalTableScanRule(final PlannerContext plannerContext)
   @Override
   public boolean matches(RelOptRuleCall call)
   {
-    if (plannerContext.getQueryMaker().feature(QueryFeature.CAN_READ_EXTERNAL_DATA)) {
+    if (plannerContext.engineHasFeature(EngineFeature.CAN_READ_EXTERNAL_DATA)) {
       return super.matches(call);
     } else {
-      plannerContext.setPlanningError("SQL query requires scanning external datasources that is not suported.");
+      plannerContext.setPlanningError(
+          "Cannot use '%s' with the current SQL engine.",
```

Comment:
I thought about it, but as usual in CS: naming is the hard part 🙂. I think we can add engine names in the future. I thought it would involve some discussion about what to call them in user-facing messages, so better to have a dedicated PR for that.

Reply:
I don't like including Java class names in user-facing error messages: it seems leaky. But I get why both reviewers here want to add something other than "current engine".

Maybe we can try having the debate about names now, and see if we can reach agreement quickly.

Suggestion: `native` for the current sole production engine, and `msq-task` for the task-based multi-stage-capable engine we plan to introduce in the future?

Reply:
I added a `name()` method, and went with `native` for the name.